### ISSUE TITLE
[Guided Setup] [CVP] Add animated tooltip to GBR from Concierge/ExpensifyDM when a user first signs in.

### ISSUE THREAD

**Issue by danielrvidal** [Expensify/App#42218]
Problem: When users sign up for NewDot on mobile, they are dropped into the chat list and have a lot of options to click. We have invested in onboarding to give the users instructions on what to do based on their intentions. We currently put in a GBR but even that leaves it up to chance they might not click the GBR and learn how to use the app. If we do not make it very clear what to do next upon sign-up, we risk the chance of converting the user. 

Solution: For new users on NewDot who sign up via mobile, add an animated tool tip so it is very clear what to do next. It would look like this:

![CleanShot 2024-05-14 at 07 34 29](https://github.com/Expensify/App/assets/2364487/78876325-a8ad-4ae9-9f8c-0ae7fd6cd48f)

For note, web does not need to do this because we open up and show the LHN AND have the chat open already. 

Original discussion here: https://expensify.slack.com/archives/C03TBKRSB/p1715806228356299?thread_ts=1715040142.903759&cid=C03TBKRSB

cc @anmurali @dubielzyk-expensify @trjExpensify 

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~018075b7c8b5ef5a84</li>
        <li>Upwork Job ID: 1793677041663877120</li>
        <li>Last Price Increase: 2024-05-23</li>
<li>Automatic offers: </li>
<ul>
<li>dukenv0307 | Reviewer | 102451041</li>
<li>tienifr | Contributor | 102451045</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @dukenv0307</details>

**Comment by jamesdeanexpensify** [comment]
Would love to make the tool tip shorter, just `Get started here!`

**Comment by trjExpensify** [comment]
Awesome! We're implementing the bones of this educational tooltip component for mobile in [this PR](https://github.com/Expensify/App/pull/40066) for the quick action, so hopefully it can be reusable very easily for this purpose. CC: @tienifr

**Comment by dubielzyk-expensify** [comment]
> Would love to make the tool tip shorter, just Get started here!

Yes please!

> so hopefully it can be reusable very easily for this purpose.

Yeah that was my hope too. I've added some clarifying questions to ensure we can at least adjust the positioning as that'll be different for this issue

**Comment by danielrvidal** [comment]
@tienifr the copy and issue is not ready for you to work on. Taking it off hold/planning. 

English: 
> Get started here!

Spanish:
>Comience aquí

**Comment by trjExpensify** [comment]
Going to assign this to @tienifr and take off the `Help wanted` label as it'll be a follow-up to https://github.com/Expensify/App/pull/40066

Let's still get a proposal for it reviewed by @dukenv0307 when the time is right, but let's please complete the OG PR as a priority. 👍

**Comment by trjExpensify** [comment]
Perfect, thanks. Assigned!

**Comment by danielrvidal** [comment]
Hey team, any update on when we think we'll get to this one?

**Comment by danielrvidal** [comment]
@dukenv0307 @tienifr what is the update on this one?

**Comment by danielrvidal** [comment]
Got it, makes sense. Thank you for the update!

**Comment by danielrvidal** [comment]
Just noting that we're still waiting on @tienifr to finished the issue linked early so we're holding.

**Comment by danielrvidal** [comment]
Not sure why Melvin is freaking out here. We're still on hold.

**Comment by danielrvidal** [comment]
@tienifr is still [working on the QAB tooltip](https://github.com/Expensify/App/pull/40066), so we're still holding until that is done.

**Comment by danielrvidal** [comment]
Woo! @tienifr I saw the other tooltip merge yesterday, so sounds like we should be able to work on this one next?

**Comment by trjExpensify** [comment]
Removing the hold from the title!

**Comment by danielrvidal** [comment]
Awesome, thank you!

**Comment by danielrvidal** [comment]
@tienifr it looks like you are woirking on this in the PR listed above. Let us know if you run into any issues. 

Also, @trjExpensify and @anmurali were talking and I believe this should be treated as CRITICAL. I'm guessing you are already prioritizing but please let me know if you have any question.

**Comment by trjExpensify** [comment]
Might be worth bringing that to Slack @tienifr and tagging a few people you think would be good to provide input on what might be wrong. 👍

**Comment by trjExpensify** [comment]
Did you do that, @tienifr? Can you link the thread here.

**Comment by trjExpensify** [comment]
Putting this back on Daily. @tienifr please provide an update [on this](https://github.com/Expensify/App/issues/42218#issuecomment-2199761591).

**Comment by trjExpensify** [comment]
Bump again here, @tienifr. This is a deliverable for our next deadline on July 22nd. We want to get this in place in good time, can you prioritise moving the draftPR into review today? Thanks!

**Comment by trjExpensify** [comment]
Well you couldn't get better timing, ey! https://expensify.slack.com/archives/C01GTK53T8Q/p1720001303477089

**Comment by danielrvidal** [comment]
It looks like @tienifr is still working on the solution here: https://expensify.slack.com/archives/C01GTK53T8Q/p1720455400941889?thread_ts=1720001303.477089&cid=C01GTK53T8Q

@roryabraham just gave him somewhere to research, but we're still waiting on the solution so it's still a WIP.

**Comment by trjExpensify** [comment]
@Expensify/design question for yah above on this green light bulb. 

I see we didn't add that to the QAB educational tooltip ([mocks in the OP here](https://github.com/Expensify/App/issues/38054)), so do we for sure want it here? If so, can we get it to @tienifr and let's add it on the QAB tooltip as well.

**Comment by shawnborton** [comment]
Hmm good find, curious what @dubielzyk-expensify might be thinking here - I do like including it for the QAB tooltip personally, it does look quite nice when it's there.

**Comment by trjExpensify** [comment]
Yeah, sorry if that wasn't clear, I like it as well. Just noticed we don't have it on the QAB tooltip, so wanted to be sure you guys do want it, and we should add it in all places.

**Comment by dannymcclain** [comment]
>Yeah, that would be my vote - use it and be consistent everywhere 👍

Yeah I like the idea of using it everywhere. It's a nice little detail!

**Comment by trjExpensify** [comment]
Great! Anyone know where that icon is so @tienifr can use it?

**Comment by dannymcclain** [comment]
I see it here: `App/assets/images/emojiCategoryIcons/light-bulb.svg`

Is it ok to use from that directory or should we also add it to `App/assets/images` (which is where all of our regular UI icons live)?

**Comment by trjExpensify** [comment]
Oh but that one looks different than the icon in the tooltip... 

![image](https://github.com/Expensify/App/assets/16232057/4761e984-ab16-4b16-bbee-7fe3fc89622e)

<img width="389" alt="image" src="https://github.com/Expensify/App/assets/16232057/b0f92c27-3319-4b42-b32b-cc0beaee7770">

**Comment by trjExpensify** [comment]
Sorry, I was in the wrong place! https://github.com/Expensify/App/blob/main/assets/images/emojiCategoryIcons/light-bulb.svg

**Comment by shawnborton** [comment]
I have no idea where the icon _should_ be placed... I don't think there's a great reason why it ended up in the emojiCategory folder, we were probably just trying to organize things or something?

**Comment by dannymcclain** [comment]
It's used as a category icon in the emoji picker (which is why I think it's in the emojiCategory) folder. So that makes sense, I just don't know if it's cool to use it from there for other places. 🤷‍♂️ I'll let the engineers weigh in on that.

**Comment by dubielzyk-expensify** [comment]
I've had a few versions tossing around with and without icon. I could see icon being a optional prop, but happy to use it and be consistent everywhere 👍

**Comment by danielrvidal** [comment]
So does that give @tienifr everything he needs?

**Comment by danielrvidal** [comment]
@tienifr can you please provide an update on this one?

**Comment by trjExpensify** [comment]
@dukenv0307 I'm assigning you to the draftPR so you can participate. We would very much appreciate the focus on getting this over the line. It's dragging out. Thanks!

**Comment by danielrvidal** [comment]
@trjExpensify[ bumped the draftPR](https://github.com/Expensify/App/pull/44277) one today for @tienifr so we're still making progress.

**Comment by trjExpensify** [comment]
Putting this back on daily.

**Comment by quinthar** [comment]
Saw this way above:
> For note, web does not need to do this because we open up and show the LHN AND have the chat open already.
This should work on mobile web, but maybe not on desktop.

**Comment by francoisl** [comment]
> @danielrvidal Can we use Onyx's `nvp_isFirstTimeNewExpensifyUser` to indicate first-time sign up users? I tried with a completely new email but the BE still returns `nvp_isFirstTimeNewExpensifyUser` as false.

@tienifr You can use that NVP. The wrong value you get might be due to a rerender because of Strict Mode in dev. `OpenApp` gets called twice, which causes the NVP to be set to `false`. Try following https://expensify.slack.com/archives/C01GTK53T8Q/p1720173674628659 to test with strict mode disabled.

**Comment by trjExpensify** [comment]
I can repro that as well on staging, FWIW: 

https://github.com/user-attachments/assets/840f90d2-52a6-450d-9b7f-8467abb132ad

**Comment by francoisl** [comment]
I can repro on staging as well, looks like it's being updated to `false` by `GetMissingOnyxMessages`. The way this NVP is updated in the backend is not ideal at the moment, though I don't have a great solution for you in mind at the moment. I'm asking around internally to see if there's a more reliable way to make this work.

<img width="1512" alt="image" src="https://github.com/user-attachments/assets/7251972c-d4c8-4c65-8950-d33bf28080ce">

**Comment by francoisl** [comment]
Hi @dukenv0307, we have an internal tracking issue for a backend fix, though there's no movement on it just yet. I should be able to get to it ~next week if no one else can pick it up by then.

**Comment by trjExpensify** [comment]
@marcaaron is working on it, hopefully in the next couple of days it'll be ready.

**Comment by trjExpensify** [comment]
Copying these BE blockers over for my tracking: 

https://github.com/Expensify/Auth/pull/12236 - merged, awaiting a deploy
https://github.com/Expensify/Web-Expensify/pull/43274 - held on auth above.

**Comment by trjExpensify** [comment]
Web is merged, awaiting a deploy.

**Comment by trjExpensify** [comment]
We're still waiting on today's Web deploy to get this to staging. Our Web deployer is on PST this week, so likely to go out before end of day on the US west coast. 👍

**Comment by trjExpensify** [comment]
Web PR is on staging.

**Comment by marcaaron** [comment]
Changes should be live on production now.

**Comment by mallenexpensify** [comment]
Contributor: @tienifr owed $250 via NewDot
Contributor+: @dukenv0307 paid $250 via Upwork

Do we want a regression test here?

**Comment by mallenexpensify** [comment]
There's a test case here https://expensify.testrail.io/index.php?/cases/view/3272232
![image](https://github.com/user-attachments/assets/c772e558-36ee-4c7a-b2bf-c7f46040a616)

Closing this out.  Thanks @dukenv0307 and @tienifr , sorry payment took a bit (automation _should_ be working correctly now)

**Comment by danielrvidal** [comment]
Thanks @mallenexpensify!

**Comment by garrettmknight** [comment]
$250 approved for @tienifr
